app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java

```diff
@@ -19,12 +21,16 @@
     public void onCreate() {
         super.onCreate();
         Fabric.with(this, new Crashlytics());
-        CollectionServiceStarter collectionServiceStarter = new CollectionServiceStarter(getApplicationContext());
-        collectionServiceStarter.start(getApplicationContext());
+        CollectionServiceStarter collectionServiceStarter = new CollectionServiceStarter(this);
+        collectionServiceStarter.start(this);
         PreferenceManager.setDefaultValues(this, R.xml.pref_general, false);
         PreferenceManager.setDefaultValues(this, R.xml.pref_data_sync, false);
         PreferenceManager.setDefaultValues(this, R.xml.pref_notifications, false);
         PreferenceManager.setDefaultValues(this, R.xml.pref_data_source, false);
-        new IdempotentMigrations(getApplicationContext()).performAll();
+        new IdempotentMigrations(this).performAll();
+
+        //create index
+        SQLiteDatabase db = Cache.openDatabase();
+        db.execSQL("CREATE UNIQUE INDEX IF NOT EXISTS bgtimestampindex ON bgreadings(timestamp)");
```

Comment:
You might want to experiment with `CREATE UNIQUE INDEX IF NOT EXISTS bgtimestampindex ON bgreadings(timestamp DESC)` as well.  I'm not sure which will perform better, if either.  I guess a separate index to get it going both ways is an option too.